hardware/raspberrypi/bootmodes/msd.md

```diff
@@ -2,9 +2,22 @@
 
 **USB mass storage boot is available on Raspberry Pi 2B v1.2, 3A+, 3B, and 3B+ only. Support for USB mass storage boot will be added to the Raspberry Pi 4B in a future software update.**
 
-This tutorial explains how to boot your Raspberry Pi from a USB mass storage device such as a flash drive or a USB hard disk. Note that this feature does not work with all USB mass storage devices.
+This page explains how to boot your Raspberry Pi from a USB mass storage device such as a flash drive or a USB hard disk. Note that this feature does not work with all USB mass storage devices.
 
-## Program USB host boot mode (Raspberry Pi 3A+ and 3B only)
+See the [bootmodes documentation](README.md) for the boot sequence and alternative boot modes (Network, USB device, GPIO or SD boot).
+
+Note that "USB mass storage boot" is different from "USB device boot mode". The [device boot mode](device.md) allows a Raspberry Pi connected to a computer to boot using files on that computer.
+
+For devices that are not supported, an alternative is to use the 'special bootcode.bin-only boot mode' as described [here](README.md). This still requires/boots from an SD-card, but `bootcode.bin` is the only file read from the SD-Card.
+
+[Pi 2B v1.2 up to 3](#pi2to3)
+
+[Pi 3B+ CM3+](#pi3plus)
```

Comment:
I think these should be removed. It's not obvious they refer to stuff further down the page, and they're not really necessary anyway.